File: src/main/java/de/floribe2000/warships_java/direct/api/stats/StatsContainer.java
package de.floribe2000.warships_java.direct.api.stats;

public class StatsContainer extends BaseStatsContainer {

	/**
	 * max frags in a battle
	 */
	private int max_frags_battle;

	/**
	 * total draws
	 */
	private int draws;

	/**
	 * max xp in a battle
	 */
	private int max_xp;

	/**
	 * total wins
	 */
	private int wins;

	/**
	 * total planes killed
	 */
	private int planes_killed;

	/**
	 * total losses
	 */
	private int losses;

	/**
	 * torpedo stats
	 */
	private ExtendedWeaponStatsWithShip torpedoes;

	/**
	 * total battles
	 */
	private int battles;

	/**
	 * max damage in a battle
	 */
	private int max_damage_dealt;

	/**
	 * total damage
	 */
	private int damage_dealt;

	/**
	 * max plane kills in a battle
	 */
	private int max_planes_killed;

	/**
	 * aircraft stats
	 */
	private WeaponStatsWithShip aircraft;

	/**
	 * ramming stats
	 */
	private WeaponStatsWithShip ramming;

	/**
	 * main battery stats
	 */
	private ExtendedWeaponStatsWithShip main_battery;

	/**
	 * secondary battery stats
	 */
	private ExtendedWeaponStatsWithShip secondary_battery;

	/**
	 * total survived wins
	 */
	private int survived_wins;

	/**
	 * total frags
	 */
	private int frags;

	/**
	 * gained xp
	 */
	private int xp;

	/**
	 * total survived battles
	 */
	private int survived_battles;

}

File: src/main/java/de/floribe2000/warships_java/direct/api/stats/StatsContainer.kt
package de.floribe2000.warships_java.direct.api.stats

class StatsContainer : BaseStatsContainer() {
    /**
     * max frags in a battle
     */
    val max_frags_battle = 0

    /**
     * total draws
     */
    val draws = 0

    /**
     * max xp in a battle
     */
    val max_xp = 0

    /**
     * total wins
     */
    override val wins = 0

    /**
     * total planes killed
     */
    val planes_killed = 0

    /**
     * total losses
     */
    override val losses = 0

    /**
     * torpedo stats
     */
    val torpedoes: ExtendedWeaponStatsWithShip? = null

    /**
     * total battles
     */
    override val battles = 0

    /**
     * max damage in a battle
     */
    val max_damage_dealt = 0

    /**
     * total damage
     */
    val damage_dealt = 0

    /**
     * max plane kills in a battle
     */
    val max_planes_killed = 0

    /**
     * aircraft stats
     */
    val aircraft: WeaponStatsWithShip? = null

    /**
     * ramming stats
     */
    val ramming: WeaponStatsWithShip? = null

    /**
     * main battery stats
     */
    val main_battery: ExtendedWeaponStatsWithShip? = null

    /**
     * secondary battery stats
     */
    val secondary_battery: ExtendedWeaponStatsWithShip? = null

    /**
     * total survived wins
     */
    override val survived_wins = 0

    /**
     * total frags
     */
    val frags = 0

    /**
     * gained xp
     */
    override val xp: Long = 0

    /**
     * total survived battles
     */
    override val survived_battles = 0
}